jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+
+        self.table.headers_avail = ["Title", "Instrument", "DateObs", "URL"]
+        self.table.headers_visible = ["Title", "Instrument", "DateObs"]
+        self._populate_url_only = False
+
+        self.table.show_rowselect = True
+        self.table.item_key = "URL"
+
+        self.hub.subscribe(self, AddDataMessage, handler=self.vue_center_on_data)
+        self.hub.subscribe(self, RemoveDataMessage, handler=self.vue_center_on_data)
+        self.hub.subscribe(self, LinkUpdatedMessage, handler=self.vue_center_on_data)
+
+    @observe("viewer_selected", type="change")
+    def vue_viewer_changed(self, _=None):
+        # Check mixin object initialized
+        if hasattr(self, "viewer"):
+            # Clear all existing subscriptions and resubscribe to selected viewer
+            for viewer in self.viewer.viewers:
+                if viewer == self.viewer.selected_obj:
+                    viewer.state.add_callback("zoom_center_x", self.vue_center_on_data)
+                    viewer.state.add_callback("zoom_center_y", self.vue_center_on_data)
+                else:
+                    # If not subscribed anyways, remove_callback should produce a no-op
+                    viewer.state.remove_callback(
+                        "zoom_center_x", self.vue_center_on_data
+                    )
+                    viewer.state.remove_callback(
+                        "zoom_center_y", self.vue_center_on_data
+                    )
+            self.vue_center_on_data()
+
+    @observe("coord_follow_viewer_pan", type="change")
+    def _toggle_viewer_pan_tracking(self, _=None):
+        """Detects when live viewer tracking toggle is clicked and centers on data if necessary"""
+        # Center on data if we're enabling the toggle
+        if self.coord_follow_viewer_pan:
+            self.vue_center_on_data()
+
+    def vue_center_on_data(self, event=None):
+        """
+        If data is present in the default viewer, center the plugin's coordinates on
+        the viewer's center WCS coordinates.
+        """
+        if not hasattr(self, "viewer"):
+            # mixin object not yet initialized
+            return
+        # If plugin is in "Manual" mode, we should never
+        # autocenter and potentially wipe the user's data
+        if self.viewer_selected == "Manual":
+            return
+
+        # If the user panned but tracking not enabled, don't recenter
+        # NOTE: float event typecheck assumes the only "float" type event triggering this method
+        #   is "zoom_center_x" and "zoom_center_y"
+        #   Otherwise, need some other way to determine the event = user panning
+        if isinstance(event, float) and not self.coord_follow_viewer_pan:
+            # Thus, we're no longer centered
+            self.viewer_centered = False
+            return
+
+        # gets the current viewer
+        if self.viewer_selected:
+            viewer = self.viewer.selected_obj
+
```

Comment:
if the if-statement isn't entered, `viewer` seems to be undefined and likely would raise an error in the next line.  Maybe instead, `self.viewer_selected == ''` should return along with "Manual" at the top and then this can be pulled out of the if-statement?

Reply:
Good catch! Fixed in https://github.com/spacetelescope/jdaviz/pull/2872/commits/a2b889098f43dfa9e181224651d4f308d358fccf